src/cargo/util/progress.rs

```diff
@@ -23,8 +26,8 @@ struct Throttle {
     last_update: Instant,
 }
 
-struct State<'cfg> {
-    config: &'cfg Config,
```

Comment:
> This also fixes a long-standing issue of transfer speeds never heading
to zero if git2 stops calling the progress function.

I think saying this is a mistake as the statement preempts the progress implementation with `gitoxide` which indeed doesn't have this issue. The `git2` callback-based progress didn't changed, so the problem would persist there.

The `gitoxide` implementation doesn't get starved in terms of callbacks due to the progress bare polling the progress state in set intervals of 300ms (like it was before, I think). But if `gitoxide` would implement a callback mechanism (which it could), thread-safety wouldn't be needed here anymore, but it could starve progress just like `git2` does.

Thus I am advocating for keeping this approach, and thus the thread safety that was added in the commit.

Reply:
> Thus I am advocating for keeping this approach, and thus the thread safety that was added in the commit.

Sorry, my intent wasn't to discourage this work but to see if it could be reviewed and merged first to keep both PRs more focused.

Reply:
Ah, right! I think it makes sense for it to start out here to see it in context, and if you agree it's a necessity to move the `gitoxide` integration forward, I am happy to move the commit into its own PR. I updated my TODO list accordingly.